doc/SAI-IPv6-Segment-Routing-VPN.md

```diff
@@ -0,0 +1,674 @@
+
+### No duplication of SID Lists and SRv6 NHs for every VPN
+In current SAI SRv6 specification, the SRv6 NHs are configured using SRv6 Tunnel OID and a SID List OID.
+When multiple VPN SID's exist behind an end node, NOS would need to create
+several SID Lists, one per VPN SID. SID Lists with VPN SID would be {<v6
+address SID List>, VPN SID}. As a result, the number of SRv6 SID Lists (and
+NH's) required to support all endpoints with the VPN SID would be **# VPN
+SID's** * **underlay SRv6 paths**
+
+Also any change in underlay SRv6 path would require NOS to update **VPN SID's** * SRv6 SID Lists.
+
+### Statistics per policy
+Allow statistics collection per SID List (per path). This allows SONiC/NOS to collect statistics per policy.
+
+### L3VPN only
+
+L3VPN only use case is when SRv6 is used to provide SRv6 L3VPN solution while
+the underlay forwarding protocol are segregated, and hence the PE would not
+impose the entire SRv6 path. This allows deployment of SRv6 solution as the
+network is being upgraded to be SRv6 aware.
+
+![SRv6 L3VPN topology](figures/SRv6_l3vpn_topology.png "Figure 1: SRv6 L3 VPN topology ")
+__Figure 1: SRv6 L3VPN Topology.__
+
+Traffic to 12/8 network/VPN NODE21, would have below header.
+
+```
+   Outer IPv6 DA: FD00:201:XXX:FFF1:A/80 (VPN_SID_NODE21,ORANGE)
+```
+
+## SAI SRv6 Pipeline
+![SAI SRv6 Pipeline](figures/srv6_vpn_pipeline.png "Figure 1: SRv6 pipeline")
+__Figure 1: SRv6 pipeline__
+
+## SAI Changes
+
+### Prefix Aggregation ID
+Prefix Aggregation ID is a number that identifies a target on the overlay
+network, such as a VPN service. The route can be associated to a single SRv6
+Next Hop (Endpoint) or a Next hop group of SRv6 NH's for the VPN. Prefix
+Aggregation Id allows SAI to derive VPN SID per SRv6 Endpoint (SRv6 Tunnel)
+when muliple SRv6 Endpoints are part of the VPN service.
+
+The tunnel map entry table is programmed with the prefix aggregation id to VPN SID mapping.
+
+### NULL SIDList / SIDList with empty list
+L3VPN solution is achieved by allowing SIDLIST OID to be NULL or creating a SIDLIST OID with empty sidlist array.
+SIDList with empty sidlist array could be used for cases where the encapsulated packets need to be directed to a specific underlay NH.
+
+### Tunnel
+Every SRv6 End node is represented by a SAI P2P Tunnel entry. All the VPN SIDs on this end node are specified using a tunnel map.
+
+### Tunnel Map
+Tunnel Map entry table contains mapping of Prefix aggregation ID to VPN SID of the remote end node.
+
+### SAI Object relationship
+
+The diagram below shows the objects and relationship. The diagram only shows relevant fields in the different tables.
+
+NOTE: Below object types have been combined to keep the diagram simple.
+* Next hop group and member list
+* Tunnel map and Tunnel map entry table
+
+New objects are highlighed in RED.
+
+![SAI SRv6 Encap model](figures/sai_srv6_encap_model.png "Figure 1: SRv6 VPN Encap model ")
```

Comment:
SIDLIST OID field from SRv6 NH object should be pointing to an object. why is it pointing to content of an object?

Reply:
Thank you for your comments. I've updated the diagram. Hopefully its clearer now.

Reply:
updated the diagram